Content.Server/WebPlacer/WebPlacerSystem.cs

```diff
@@ -0,0 +1,102 @@
+using Content.Server.Popups;
+using Content.Shared.Maps;
+using Content.Shared.WebPlacer;
+using Content.Shared.Whitelist;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+
+namespace Content.Server.WebPlacer;
+
+/// <summary>
+/// Spawns entities (probably webs) around the component owner. Action handled by <see cref="SharedWebPlacerSystem"/>.
+/// </summary>
+public sealed class WebPlacerSystem : SharedWebPlacerSystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelistSystem = default!;
+    [Dependency] private readonly ITileDefinitionManager _tile = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedMapSystem _map = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<WebPlacerComponent, SpiderWebActionEvent>(OnSpawnNet);
+    }
+
+    private void OnSpawnNet(EntityUid uid, WebPlacerComponent component, SpiderWebActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var transform = Transform(uid);
+        var grid = transform.GridUid;
+
+        // Instantly fail in space
+        if (grid == null || !TryComp<MapGridComponent>(grid, out var gridComp))
+        {
+            _popup.PopupEntity(Loc.GetString(component.MessageOffGrid), args.Performer, args.Performer);
+            return;
+        }
+
+        var coords = transform.Coordinates;
+        List<EntityCoordinates> spawnPos = new();
+
+        var vects = new List<Vector2i>();
+        vects.AddRange(component.OffsetVectors);
+
+        foreach (var vect in vects)
+            spawnPos.Add(coords.Offset(vect));
```

Comment:
The local `vects` variable seems to just copy `component.OffsetVectors` and nothing else is added/mutated.

Is there a reason we can't just use `component.OffsetVectors` directly?

```suggestion
        foreach (var vect in component.OffsetVectors)
            spawnPos.Add(coords.Offset(vect));
```